src/rashdf/plan.py

```diff
@@ -1117,6 +1121,106 @@ def reference_lines_timeseries_output(self) -> xr.Dataset:
         """
         return self.reference_timeseries_output(reftype="lines")
 
+    def observed_timeseries_input(self, vartype: str = "Flow") -> xr.Dataset:
+        """Return observed timeseries input data for reference lines and points from a HEC-RAS HDF plan file.
+
+        Parameters
+        ----------
+        vartype : str, optional
+            The type of observed data to retrieve. Must be either "Flow" or "Stage".
+            (default: "Flow")
+
+        Returns
+        -------
+        Dict[Site: str, xr.DataArray]
+            An xarray Dataset with observed timeseries data for both reference lines and reference points.
+
+            Coordinates:
+            - 'time': datetime64[ns]
+
+            Data variables:
+            - 'Flow' or 'Stage': float64
+        """
+
+        # Decode the contents of the DataFrame from utf-8
+        def decode_bytes(val):
+            if isinstance(val, bytes):
+                return val.decode("utf-8")
+            return val
+
+        if vartype == "Flow":
+            output_path = self.OBS_FLOW_OUTPUT_PATH
+        elif vartype == "Stage":
+            output_path = self.OBS_STAGE_OUTPUT_PATH
+        else:
+            raise ValueError('vartype must be either "Flow" or "Stage".')
+
+        observed_group = self.get(output_path)
+        if observed_group is None:
+            raise RasPlanHdfError(
+                f"Could not find HDF group at path '{output_path}'."
+                f" Does the Plan HDF file contain reference {vartype[:-1]} output data?"
+            )
+        print(f"observed_group: {observed_group}")
+        if "Attributes" in observed_group.keys():
+            attr_path = observed_group["Attributes"]
+            print(f"attr_path: {attr_path}")
+            attrs_df = pd.DataFrame(attr_path[:])
+            # Apply the decoding function to each element in the DataFrame
+            attrs_df = attrs_df.map(decode_bytes)
+            if vartype == "Flow":
+                attrs_df["Units"] = "cfs"
+            elif vartype == "Stage":
+                attrs_df["Units"] = "ft"
+            else:
+                attrs_df["Units"] = "Unknown"
```

Comment:
I think we need to leave `Units` out of this since it's not explicitly set anywhere in the "Event Conditions" part of the HDF file, as far as I can tell. If we really want to set `Units` there's a top level "Units System" attribute in the HDF which could be used to infer the units here. I realize that assuming `cfs` and `ft` is pretty safe in our use case, but I think we should aim for a higher level of compatibility. :)

Reply:
Units now left out of this process